tests/test_model.py

```diff
@@ -344,6 +344,23 @@ def test_sum_composite_models(self):
         for mod in [model1, model2, model3, model4]:
             self.assertTrue(mod in model_total3.components)
 
+    def test_hints_in_composite_models(self):
+        # test propagation of hints from base models to composite model
+        def func(x, amplitude):
+            pass
+
+        m1 = Model(func, prefix='p1_')
+        m2 = Model(func, prefix='p2_')
+
+        m1.set_param_hint('amplitude', value=1)
+        m2.set_param_hint('amplitude', value=2)
+
+        mx = (m1 + m2)
+        params = mx.make_params()
+        param_values = {name: p.value for name, p in params.items()}
+        self.assertTrue(param_values['p1_amplitude'] == 1)
+        self.assertTrue(param_values['p2_amplitude'] == 2)
```

Comment:
Just FYI, `self.assertEqual` would be better here. It gives a more useful error message when violated, printing the actual and expected values.